_wzl/wzl.py

```diff
@@ -0,0 +1,391 @@
+
+    \b
+     - Copy example config files.
+    """
+    config_dir = 'config'
+    config_files = set(os.listdir(config_dir))
+    for f in config_files:
+        prefix, sep, suffix = f.rpartition('.example')
+        if sep and not suffix and prefix not in config_files:
+            click.echo('{} => {}'.format(f, prefix))
+            shutil.copy(
+                os.path.join(config_dir, f), os.path.join(config_dir, prefix))
+
+
+@wzl.command()
+@click.option('-d', '--detach', is_flag=True, help=(
+    "Don't run the server in the foreground."
+))
+def run(detach):
+    """
+    Start the app server.
+
+    This requires all of the base images to have been built.
+    """
+    if is_wzl_dev():
+        detach_flag = '-d' if detach else '--abort-on-container-exit'
+        forward([
+            'docker-compose', 'up', detach_flag, '--remove-orphans',
+        ])
+    else:
+        forward([
+            'twistd', '--pidfile=', '-n',
+            '-y', pkg_resources.resource_filename('weasyl', 'weasyl.tac'),
+        ])
+
+
+@wzl.command()
+@ensure_wzl_dev
+def url():
+    """
+    Get the URL for the running weasyl instance.
+
+    This will be nginx's port, but that will reverse-proxy to all the right
+    places.
+    """
+    nginx = subprocess.check_output(['docker-compose', 'port', 'nginx', '80'])
+    ip, _, port = nginx.strip().partition(':')
+    if ip == '0.0.0.0':
+        host = os.environ.get('DOCKER_HOST')
+        if host is None:
+            ip = '127.0.0.1'
+        else:
+            parsed = urlparse.urlparse(host)
+            ip, _, _ = parsed.netloc.partition(':')
+    click.echo('http://{}:{}'.format(ip, port))
+
+
+def can_connect(host, port):
+    s = socket.socket()
+    s.settimeout(5)
+    while True:
+        try:
+            s.connect((host, port))
+        except socket.error as e:
+            if e.errno in (errno.ECONNREFUSED, errno.ETIMEDOUT):
+                return False
+            elif e.errno == errno.EINTR:
+                continue
+            else:
+                raise
+        else:
+            return True
+
+
+def wait_for_postgres():
+    started_at = time.time()
+    if can_connect('db', 5432):
+        return
+
+    bar = click.progressbar(
+        length=60,
+        show_percent=False, show_eta=False,
+        label='waiting for postgres to come up')
+    with bar:
+        while True:
+            waited = time.time() - started_at
+            if waited > 60:
+                raise click.Abort("waited too long for postgres to come up")
+            bar.update(waited)
+            time.sleep(1)
+            if can_connect('db', 5432):
+                return
+
+
+@wzl.command(add_help_option=False, context_settings=dict(
+    ignore_unknown_options=True,
+))
+@click.argument('args', nargs=-1, type=click.UNPROCESSED)
+@forward_from_wzl_dev
+def test(args):
+    """
+    Run the tests.
+
+    This will also wait until postgres comes up before beginning.
+    """
+
```

Comment:
I have mixed feelings about whether to include it or not. I can't decide, so maybe someone else should.